Modules/_ctypes/_ctypes.c

```diff
@@ -967,18 +967,30 @@ CDataType_in_dll_impl(PyObject *type, PyTypeObject *cls, PyObject *dll,
         return NULL;
     }
 #else
+    dlerror();
```

Comment:
I'm not a huge fan of this workaround, but I guess it's simple enough. It would be nice to emit a warning (or possibly raise an error early) if this fails. Also, could you add a comment with the issue number explaining why this is needed, so we don't confuse the future readers?

Reply:
ACK, I'll add a comment explaining the rationale.